tests/test_integration.py

```diff
@@ -1,37 +1,111 @@
 import json
+import os
 from pathlib import Path
 from tempfile import TemporaryDirectory
-from emanate import main
+from contextlib import contextmanager
+import emanate
 
 
-def test_config_relative_path():
-    """Test paths relative to the configuration file.
+def main(*pargs):
+    args = map(lambda x: x if isinstance(x, str) else str(x), pargs)
+    emanate.main(args)
 
-    Creates the following hierarchy, and calls `emanate --source tmpdir/src`:
-    tmpdir
-    ├── dest
-    └── src
-        ├── foo
-        └── emanate.json
 
-    This test then asserts that tmpdir/dest/foo exists, and that
-    tmpdir/dest/emanate.json does not.
-    """
+@contextmanager
+def cd(path):
+    if isinstance(path, Path):
+        path = str(path)
+
+    cwd = Path.cwd()
+    try:
+        os.chdir(path)
+        yield
+    finally:
+        os.chdir(str(cwd))
+
+
+@contextmanager
+def directory_tree(obj):
+    def mktree(path, obj):
+        assert isinstance(path, Path)
+        # File by content
+        if isinstance(obj, str):
+            with path.open(mode='w') as file:
+                file.write(obj)
+
+        # Directory
+        elif 'type' not in obj:
+            path.mkdir(exist_ok=True)
+            for name, child in obj.items():
+                mktree(path / name, child)
+
+        elif obj['type'] == 'link':
+            path.symlink_to(obj['target'])
+
     with TemporaryDirectory() as tmpdir:
-        # Setup the test environment
         tmpdir = Path(tmpdir)
-        src = tmpdir.joinpath('src')
-        dest = tmpdir.joinpath('dest')
-        src.mkdir()
-        src.joinpath('foo').touch()
-        dest.mkdir()
-
-        with src.joinpath('emanate.json').open(mode='w') as config:
-            json.dump({
-                'destination': '../dest',
-            }, config)
-
-        assert not dest.joinpath('foo').exists()
-        main(['--source', str(src)])
-        assert dest.joinpath('foo').samefile(src.joinpath('foo'))
-        assert not dest.joinpath('emanate.json').exists()
+        mktree(tmpdir, obj)
+        yield tmpdir
+
+
+def helper(tree={}, source='src', options=lambda _: []):
```

Comment:
This needs a better name than `helper`, and it cannot start with `test` lest `pytest` try to run it.